### PR TITLE
Prompt tuning loss mask fix

### PR DESCRIPTION
**Pull request by vadam5**
Added loss mask configurability to include or exclude the text portion of the input in the prompt tuning loss calculation. 

```
[soft` prompt][Input Text][Answer][EOD Padding]
[000000000000011111111111111111111000000000000]
```
vs 

```
[soft prompt][Input Text][Answer][EOD Padding]
[00000000000000000000000011111111000000000000]
```

I also added prompt tuning CI tests and prompt tuning dataset unit tests that get called during CI testing. 

Calculating the loss on the answer only helped. The 2nd score is from calculating the loss on the full input text and the 3rd is from calculating the loss on the answer only. These scores are on Winogrande formulated as fill in the blank instead of sentence completion. (The GPT eval scores in the spread sheet are for Winogrande as sentence completion. We couldn't find a Winogrande sentence completion training set split to use for prompt tuning, so I added Winogrande fill in the blank to the model eval code on gitlab.)

Allen AI Winogrande
Baseline 1.3B Megatron GPT: 0.489345
PromptTuned 1.3B Megatron GPT Full Text Loss: 0.486977
PromptTuned 1.3B Megatron GPT Answer Only Loss: 0.528019

### CODE REVIEW

**Comment by okuchaiev** [comment]
@khcs can you please have a look? could be useful for what you are trying to do as well.

**Review by purn3ndu — DISMISSED** [review]
Reviewed. The code looks fine to me. Is this giving better results though?

**Comment by vadam5** [comment]
> Reviewed. The code looks fine to me. Is this giving better results though?

Yes it helped, the 2nd score is from calculating the loss on the full input text and the 3rd is from calculating the loss on the answer only. These scores are on Winogrande formulated as fill in the blank instead of sentence completion like with the GPT eval scores that are online. (We couldn't find a Winogrande sentence completion training set split, so I added Winogrande fill in the blank to the model eval code on gitlab. )

Allen AI Winogrande
Baseline 1.3B Megatron GPT: 0.489345
PromptTuned 1.3B Megatron GPT Full Text Loss: 0.486977
PromptTuned 1.3B Megatron GPT Answer Only Loss: 0.528019

**Review by purn3ndu — APPROVED** [review]
LGTM
